internal/types/path.go

```diff
@@ -172,3 +183,50 @@ func removeByPath(v any, path Path) {
 		// no such path: scalar value
 	}
 }
+
+// setByPath sets value by given path for both Document and Array types.
+func setByPath[T CompositeTypeInterface](comp T, path Path, value any) error {
+	innerComp, err := comp.GetByPath(path.TrimSuffix())
+	if err != nil {
+		return err
+	}
+
+	switch inner := innerComp.(type) {
+	case *Document:
+		err := inner.Set(path.Suffix(), value)
+		if err != nil {
+			return err
+		}
+	case *Array:
+		index, err := strconv.Atoi(path.Suffix())
+		if err != nil {
+			return err
+		}
+		err = inner.Set(index, value)
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("can't set value for %T type", inner)
```

Comment:
Is this case possible at all? If yes, let's write a test (in integration tests). If not, let's panic?

Reply:
I think it's not possible, changed it to panic.